core/storage/postgres.go

```diff
@@ -0,0 +1,135 @@
+package storage
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	"time"
+
+	"github.com/kislerdm/diagramastext/core"
+	_ "github.com/lib/pq"
+)
+
+type client struct {
+	c dbClient
+}
+
+func (c client) Close(ctx context.Context) error {
+	return c.c.Close()
+}
+
+const tableWritePrompt = "public.user_prompt"
+
+func (c client) WritePrompt(ctx context.Context, v core.UserInput) error {
+	if v.UserID == "" {
+		return errors.New("user_id is required")
+	}
+	if v.RequestID == "" {
+		return errors.New("request_id is required")
+	}
+	if v.Prompt == "" {
+		return errors.New("prompt is required")
+	}
+	if v.Timestamp.Second() <= 0 {
+		v.Timestamp = time.Now().UTC()
+	}
+
+	_, err := c.c.ExecContext(
+		ctx, `INSERT INTO "`+tableWritePrompt+
+			`" ("request_id", "user_id", "prompt", "timestamp") VALUES ($1, $2, $3, $4);`,
+		v.RequestID,
+		v.UserID,
+		v.Prompt,
+		v.Timestamp,
+	)
+	return err
+}
+
+const tableWriteModelPrediction = "public.openai_response"
+
+func (c client) WriteModelPrediction(ctx context.Context, v core.ModelOutput) error {
+	if v.UserID == "" {
+		return errors.New("user_id is required")
```

Comment:
**question** why not use sentinel errors to make these errors easier to check from the caller's perspective? https://dave.cheney.net/tag/errors

Reply:
It is a good proposal. Let's log it and come back at refactoring.

For context: 
- the error is "private", end user will not see it;
- I intentionally didn't want to make commitment to a certain "error model" yet.